New test exp. "atm_gray" moved from "verification_other"

jm-c opened the pull request:
## What changes does this PR introduce?
move test experiment "atm_gray", previously in https://github.com/MITgcm/verification_other ,
into `verification`.

## What is the current behaviour? 
no test exp. using `pkg/atm_phys`

## What is the new behaviour 
bring this exp. (previously in `verification_other`) to be 1) more accessible and visible ; 2) tested over  more platforms.

## Does this PR introduce a breaking change? 
no.

## Other information:

## Suggested addition to `tag-index`
o verification/atm_gray:
  - gray atmosphere test exp moved from "verification_other".

jm-c commented:
I suggest, in this PR,  not to update docs regarding the list of experiments but rather do it in PR #885

mjlosch commented on verification/atm_gray/build/genmake_local:
```diff
@@ -0,0 +1,16 @@
+
+#  This is the local options file for the "new" version of genmake
+
+retv=1
+if test "x$OPTFILE" != x ; then
+  basename $OPTFILE | grep gfortran > /dev/null 2>&1 ; retv=$? ;
+fi
+if test $retv = 0 ; then
+  FFLAGS='-fdefault-real-8 -fdefault-double-8'
+  echo "    local gfortran setting: FFLAGS='$FFLAGS'"
+else
+  FFLAGS='-r8'
+  echo "    local default setting: FFLAGS='$FFLAGS'"
+fi
```
This only works when I specifically supply a build_options file. I.e.
```
../../../tools/genmake2 -mods ../code -of ../../../tools/build_options/darwin_arm64_gfortran 
```
gives me a working `Makefile`, but
```
../../../tools/genmake2 -mods ../code
```
does not (wrong gfortran option `-r8` leads to errors), even though the same default build options file is selected. Why do we need this here again?

mjlosch commented on verification/atm_gray/README.md:
```diff
@@ -0,0 +1,64 @@
+cubed-sphere grid set-up (6 faces 32x32, 26 levels, non uniform deltaP).
+
+### Overview:
+This experiment contains 2 aqua-planet like set-ups (with corresponding `input[.*]/` dir)
+that can be run with the same executable (built from `build/` dir using customized
+code from `code/`); binary input files have been generated using matlab script
+`gendata.m` from the `input` dir.
+Both test experiments start from a spin-up state using pickup files written after 1 year.
+
+The **primary** test, using input files from `input/` dir,
+has an interactive SST with a 10m mixed layer depth and a prescribed,
+time-invariant Q-flux. It also includes a weak damping of stratospheric winds.
+
+The **secondary** test, using files from `input.ape/` dir, is the same as the primary
+test but without stratospheric wind damping, and uses prescribed idealized SST from
+Neale and Hoskins, 2001, Aqua-Planet Experiment (APE) project.
```
Here and elsewhere: I'd prefer lines that are either `<=80 char` long or paragraphs without linebreaks. As it is now I have to resize my editor in order to avoid strange linebreaks that make it harder to read the text.

jm-c replied:
length of lines are not crazy long here, so reluctant to inforce a 80.c rule for now. And yes, I like line break at the end of a sentence, not a big if we look at it from github web page (it does not make any difference there), and when/if the file gets updated, only the affected sentence shows up in the diff (instead of the remaining part of the whole paragraph if I enforce a strict 80.c rule) which I find helpful.
In short, I would prefer to keep it this way and revisit this once we clearly agree on a rule  and start to apply this consistently (e.g., to the docs).

mjlosch commented on verification/atm_gray/input.ape/data.atm_gray:
```diff
@@ -0,0 +1,84 @@
+
+ &atmosphere_nml
+    turb = .TRUE.,
+    ldry_convection = .false.,
+    lwet_convection = .TRUE.,
+    do_virtual = .false.,
+    two_stream = .true.,
+    mixed_layer_bc = .false.,
+    roughness_heat = 0.05,
+    roughness_moist = 0.05,
+    roughness_mom = 0.05,
```
```suggestion
  turb = .TRUE.,
  ldry_convection = .false.,
  lwet_convection = .TRUE.,
  do_virtual = .false.,
  two_stream = .true.,
  mixed_layer_bc = .false.,
  roughness_heat = 0.05,
  roughness_moist = 0.05,
  roughness_mom = 0.05,
```

jm-c replied:
I just copied these parameter files from `verification_other/atm_gray`, but I agree this is not very clean so I will make these changes (+ also in `input/` version).

mjlosch reviewed (approved):
I am generally OK with this. Ignore formatting comments if you don't agree.

I am not sure about the `genmake_local` file. It kind of beats the purpose of being able to run `genmake2` (or `testreport`) without specifying a build-options file. Is that intentional?

jm-c commented:
@mjlosch I have few comment regardsing this:
> I am not sure about the genmake_local file. It kind of beats the purpose of being able to run genmake2 (or testreport) without specifying a build-options file. Is that?

1. I just took the set-up from https://github.com/MITgcm/verification_other so it's not "intentional".
2. this type of `genmake_local` has been around for some times: back in Jun 2012 I added similar `genamke_local` files for the same purpose to the 3 fizhi experiments, the commit message at the time was "a hack for this specific experiemnt (using -r8, unless gfortran is used)"  (and git blame tell me you actively contributed to make this to work with cray compiler ...).
3. You are right about the fact that it only works if I specify an optfile. And when the "getting started" section of the manual got updated (after moving to github), I told @jrscott  that it's better to alsways specify  an optfile (but I forgot a little bit why). Recently I checked again, and I think these specific `genmake_local` are likely the only reason why we should always run `testreport` with a specified optfile;  and also probably revisit the docs regarding `genmake2`.
4. I am not oppose to revisit this type of specific `genmake_local`, but since it's not just for this new test exp., it would be better to a) first agree on what to do and b) implement this in a new PR.

mjlosch commented:
By comparison to, e.g., `fizhi-cs-32x32x40`, it appears that a `.gitignore` with this content is missing in `atm_gray/build`:
```
*
!/genmake_local
```

I agree with your `genmake_local` comment: leave it for now. 
(One simple hack is to source `$gm_local` again after determining the OPTFILE, e.g. after line 1778 in `genmake2`, but I am not sure what the possible repercussion may be.)

jm-c commented:
It's easy to forget about this ".gitignore" file (as git is silent about it) so thanks @mjlosch , good catch ! Should be fixed now.

jm-c commented:
@mjlosch I just adjusted `genmake_local` here, using some of the setting form the 2 "fizhi-cs" exp. version (I cannot use exactly the same genmake_local as I also need "ALWAYS_USE_F90=1" here). With this, I think I addressed all your comments, and could think of merging this PR soon (but after 890 and 892), unless someone wants to add somthing here.

mjlosch commented:
LGTM, go ahead!

jm-c commented:
Going to merge this PR very soon.
